### PR TITLE
Factor Minmod tests into TCI tests and limiter tests

### PR DESCRIPTION
**Pull request by fmahebert**
## Proposed changes

Portions of the (old) tests that were doing detailed tests of "when and how" the limiter triggered are not grouped as tests of the TCI. Portions dealing with the packaging of neighbor data, the interaction of multiple tensors to limit, etc, remain as tests of the limiter.

Also groups remaining Minmod TEST_CASES for faster testing on clusters.

### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: tests/Unit/Evolution/DiscontinuousGalerkin/SlopeLimiters/Test_MinmodTci.cpp [hunk 1]
```diff
@@ -0,0 +1,754 @@
...
+  const size_t half_number_boundary_points = alg::accumulate(
+      alg::iota(std::array<size_t, VolumeDim>{{}}, 0_st),
+      0_st, [&mesh](const size_t state, const size_t d) noexcept {
+        return state + mesh.slice_away(d).number_of_grid_points();
+      });
+  std::unique_ptr<double[], decltype(&free)> temp_buffer(
+      static_cast<double*>(
+          malloc(sizeof(double) *
+                 (mesh.number_of_grid_points() + half_number_boundary_points))),
+      &free);
```
Comment (wthrowe) [inline tests/Unit/Evolution/DiscontinuousGalerkin/SlopeLimiters/Test_MinmodTci.cpp]: This is a test.  Unless it *really* slows things down just do the extra `VolumeDim` allocations and avoid the 15 lines of pointer-heavy code.

File: tests/Unit/Evolution/DiscontinuousGalerkin/SlopeLimiters/Test_MinmodTci.cpp [hunk 2]
```diff
@@ -0,0 +1,754 @@
...
+                          mesh.number_of_grid_points());
+  alloc_offset += mesh.number_of_grid_points();
+  std::array<DataVector, VolumeDim> boundary_buffer{};
+  for (size_t d = 0; d < VolumeDim; ++d) {
+    const size_t num_points = mesh.slice_away(d).number_of_grid_points();
+    gsl::at(boundary_buffer, d)
+        .set_data_ref(temp_buffer.get() + alloc_offset, num_points);
+    alloc_offset += num_points;
+  }
+  // Compute the slice indices once since this is (surprisingly) expensive
```
Comment (wthrowe) [inline tests/Unit/Evolution/DiscontinuousGalerkin/SlopeLimiters/Test_MinmodTci.cpp]: You only use this once, so I'm not sure what this comment is about.

File: tests/Unit/Evolution/DiscontinuousGalerkin/SlopeLimiters/Test_MinmodTci.cpp [hunk 3]
```diff
@@ -0,0 +1,754 @@
...
+
+namespace {
+
+// Helper function to wrap the preparation of the optimization buffers for the
+// troubled cell indicator function.
+template <size_t VolumeDim>
+bool wrap_allocations_and_tci(
+    const gsl::not_null<double*> u_mean,
+    const gsl::not_null<std::array<double, VolumeDim>*> u_limited_slopes,
+    const SlopeLimiters::MinmodType& minmod_type, double tvbm_constant,
```
Comment (wthrowe) [inline tests/Unit/Evolution/DiscontinuousGalerkin/SlopeLimiters/Test_MinmodTci.cpp]: `const double`

**Review by wthrowe — CHANGES_REQUESTED** [review]

**Comment by fmahebert** [comment]
Thanks for taking a look @wthrowe, fixups are posted.

**Review by kidder — DISMISSED** [review]

**Comment by kidder** [comment]
@fmahebert IWYU isn't happy

**Comment by fmahebert** [comment]
Test failure looks like #1392

**Review by nilsdeppe — COMMENTED** [review]
I'm fine with this. @wthrowe  could you please look at it when you get a chance? Thank you! :)

**Comment by wthrowe** [comment]
Looks fine.  Squash.

**Comment by fmahebert** [comment]
Squashed and rebased

**Review by wthrowe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**